tests/php/src/Order/OrderEventListenerTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace WeDevs\Dokan\Test\Order;
+
+use WeDevs\Dokan\Order\OrderEventListener;
+use WeDevs\Dokan\Test\DokanTestCase;
+
+/**
+ * @group order-event-listener
+ *
+ * @note by default the withdrawal option is enable for completed order.
+ */
+class OrderEventListenerTest extends DokanTestCase {
+
+    private $order_event_listener;
+    private $seller_id;
+
+    public function setUp(): void {
+        parent::setUp();
+        $this->order_event_listener = new OrderEventListener();
+        $this->seller_id = $this->factory()->seller->create();
+    }
+
+    public function test_processing_order_with_trash_event() {
+        $order_id = $this->create_order( 'processing' );
+        $vendor_id = dokan_get_seller_id_by_order( $order_id );
+        $initial_balance = dokan()->vendor->get( $vendor_id )->get_earnings( false );
+
+        $processing_order = wc_get_order( $order_id );
+        $processing_order->delete();
+
+        $this->order_event_listener->after_order_trash( $order_id );
```

Comment:
You should not manually trigger your implementation. It should automatically be called.

Pls follow this instruction for others.

Reply:
ok